src/Microsoft.Extensions.Logging.AzureAppServices/Internal/BlobLoggerProvider.cs

```diff
@@ -70,24 +64,18 @@ public BlobLoggerProvider(IOptionsMonitor<AzureBlobLoggerOptions> options)
 
                 var blob = _blobReferenceFactory(blobName);
 
-                Stream stream;
-                try
-                {
-                    stream = await blob.OpenWriteAsync(cancellationToken);
-                }
-                // Blob does not exist
-                catch (StorageException ex) when (ex.RequestInformation.HttpStatusCode == 404)
-                {
-                    await blob.CreateAsync(cancellationToken);
-                    stream = await blob.OpenWriteAsync(cancellationToken);
-                }
-
+                using (var stream = new MemoryStream())
                 using (var writer = new StreamWriter(stream))
                 {
                     foreach (var logEvent in eventGroup)
                     {
                         writer.Write(logEvent.Message);
                     }
+
+                    await writer.FlushAsync();
+                    var tryGetBuffer = stream.TryGetBuffer(out var buffer);
+                    Debug.Assert(tryGetBuffer);
+                    await blob.AppendAsync(buffer, cancellationToken);
```

Comment:
Is this buffer correctly scoped to the data written, or it the whole internal buffer with possible empty space at the end?